server/jetstream_api.go

```diff
@@ -287,6 +301,35 @@ type JSApiMsgDeleteResponse struct {
 
 const JSApiMsgDeleteResponseType = "io.nats.jetstream.api.v1.stream_msg_delete_response"
 
+type JSApiStreamSnapshotRequest struct {
+	// Subject to deliver the chunks to for the snapshot.
+	DeliverSubject string `json:"deliver_subject"`
+	// Do not include consumers in the snapshot.
+	NoConsumers bool `json:"no_consumers,omitempty"`
+	// Optional chunk size preference. Otherwise server selects.
+	ChunkSize int `json:"chunk_size,omitempty"`
```

Comment:
I am still at 2 minds about the max time a snapshot might run, in many industries its common to say operations interruptions are strictly time bucketed and they would rather fix what makes backups slow than have backup overrun ime, but we can always add that later

Reply:
What do you think is a sane minimum for chunk size? I'll put it in the schema you dont need to do anything but like even 1kb seems just dumb as a chunk size so I'll give some guidance in the schemas

Reply:
Normally should not be set at all. We could event remove it IMO. Server picks 64k atm.

Reply:
I updated to 128k default.